### PR TITLE
Added hosted checkout

### PR DESCRIPTION
**Pull request by ChetanGN**

### CODE REVIEW

File: woo-razorpay.php [hunk 1]
```diff
@@ -581,13 +581,41 @@ private function enqueueCheckoutScripts($data)
...
         function generateOrderForm($data)
         {
             $redirectUrl = $this->getRedirectUrl();
-            $this->enqueueCheckoutScripts($data);
+            $api = new Api($this->getSetting('key_id'), "");
+            $merchantPreferences = $api->request->request("GET", "preferences");
+
+            if(isset($merchantPreferences['options']['redirect']) && $merchantPreferences['options']['redirect'] === true)
+            {
+                Print $this->hostCheckoutScripts($data);
```
[the remark this answers is not in the document]
Comment (ChetanGN) [reply woo-razorpay.php]: I didn't get you. What is the suggestion here? @captn3m0

[the remark this answers is not in the document]
Comment (captn3m0) [reply woo-razorpay.php]: This method already returns HTML. You are changing the behaviour of the method to do a print, and then return some data.

This introduces side effects.

File: woo-razorpay.php [hunk 2]
```diff
@@ -581,13 +581,41 @@ private function enqueueCheckoutScripts($data)
...
+
         /**
          * Generates the order form
          **/
         function generateOrderForm($data)
         {
             $redirectUrl = $this->getRedirectUrl();
-            $this->enqueueCheckoutScripts($data);
+            $api = new Api($this->getSetting('key_id'), "");
+            $merchantPreferences = $api->request->request("GET", "preferences");
```
[the remark this answers is not in the document]
Comment (captn3m0) [reply woo-razorpay.php]: Can we make this a config option in the WooCommerce plugin settings instead?

[the remark this answers is not in the document]
Comment (ramth05) [reply woo-razorpay.php]: Nope, Because we don't want it to be configured and give option to merchant to only stick on standard checkout.

**Pull request merged**